src/django_components/attributes.py

```diff
@@ -2,39 +2,84 @@
 # See https://github.com/Xzya/django-web-components/blob/b43eb0c832837db939a6f8c1980334b0adfdd6e4/django_web_components/templatetags/components.py  # noqa: E501
 # And https://github.com/Xzya/django-web-components/blob/b43eb0c832837db939a6f8c1980334b0adfdd6e4/django_web_components/attributes.py  # noqa: E501
 
-from typing import Any, Dict, List, Mapping, Optional, Tuple
+from typing import Any, Dict, Mapping, Optional, Tuple
 
 from django.template import Context
 from django.utils.html import conditional_escape, format_html
 from django.utils.safestring import SafeString, mark_safe
 
 from django_components.node import BaseNode
-from django_components.util.template_tag import TagParams
-
-HTML_ATTRS_DEFAULTS_KEY = "defaults"
-HTML_ATTRS_ATTRS_KEY = "attrs"
 
 
 class HtmlAttrsNode(BaseNode):
-    def __init__(
+    """
+    Generate HTML attributes (`key="value"`), combining data from multiple sources,
+    whether its template variables or static text.
+
+    It is designed to easily merge HTML attributes passed from outside with the internal.
+    See how to in [Passing HTML attributes to components](../../guides/howto/passing_html_attrs/).
+
+    **Args:**
+
+    - `attrs` (dict, optional): Optional dictionary that holds HTML attributes. On conflict, overrides
+        values in the `default` dictionary.
+    - `default` (str, optional): Optional dictionary that holds HTML attributes. On conflict, is overriden
+        with values in the `attrs` dictionary.
+    - Any extra kwargs will be appended to the corresponding keys
+
+    The attributes in `attrs` and `defaults` are merged and resulting dict is rendered as HTML attributes
+    (`key="value"`).
+
+    Extra kwargs (`key=value`) are concatenated to existing keys. So if we have
+
+    ```python
+    attrs = {"class": "my-class"}
+    ```
+
+    Then
+
+    ```django
+    {% html_attrs attrs class="extra-class" %}
+    ```
+
+    will result in `class="my-class extra-class"`.
+
+    **Example:**
+    ```django
+    <div {% html_attrs
+        attrs
+        defaults:class="default-class"
+        class="extra-class"
+        data-id="123"
+    %}>
+    ```
+
+    renders
+
+    ```html
+    <div class="my-class extra-class" data-id="123">
+    ```
+
+    **See more usage examples in
+    [HTML attributes](../../concepts/fundamentals/html_attributes#examples-for-html_attrs).**
+    """
+
+    tag = "html_attrs"
+    end_tag = None  # inline-only
+    allowed_flags = []
+
+    def render(
         self,
-        params: TagParams,
-        node_id: Optional[str] = None,
-    ):
-        super().__init__(nodelist=None, params=params, node_id=node_id)
-
-    def render(self, context: Context) -> str:
-        append_attrs: List[Tuple[str, Any]] = []
-
-        # Resolve all data
-        args, kwargs = self.params.resolve(context)
```

Comment:
Also the `render()` methods are simpler now, because they no longer have to contain logic for figuring out what inputs the template tag received. This is now done behind the scenes by `BaseNode`. And the individual template tag implementations only need to declare the signature on the `render()` method.